mica/stats/mag_stats/catalogs.py

```diff
@@ -0,0 +1,228 @@
+       if a star catalog entry *might* have been run.  But not all approved load dirs were
+       completed so we need to also cross-correlate with obsids from telemetry.
+    """
+    global TIMELINES
+    global STARS_OBS
+    global STARS_OBS_INDICES
+    global STARS_OBS_MAP
+    global STARS_OBS_NP
+
+    # Timelines defines the mission planning `dir` of loads that were approved and
+    # at least partially run.
+
+    cmd_states_db3 = os.path.join(os.environ['SKA'], 'data', 'cmd_states', 'cmd_states.db3')
+    with DBI(server=cmd_states_db3, dbi='sqlite') as timelines_db:
+        TIMELINES = Table(timelines_db.fetchall('select dir, id from timelines '
+                                                'where datestart > "2003:007"'))
+    TIMELINES.rename_column('id', 'timeline_id')
+    TIMELINES.add_index('dir')
+    TIMELINES.add_index('timeline_id')
+
+    # Get table of every star catalog entry (a single star or fid light) that
+    # was every planned.
+
+    # The output `cat_entries` table is not useful by itself because it has
+    # multiple repeats for each entry.  The input starcheck tables only provide
+    # the source MP directory for a catalog, not the timeline ID.  Thus the output
+    # has an entry for each possible timeline_id that might correspond to the
+    # catalog.  In the next step of joining with definitive MP_STARCAT commands
+    # this all gets resolved back down.
+
+    server = os.path.join(os.environ['SKA'],
+                          'data', 'mica', 'archive', 'starcheck', 'starcheck.db3')
+    with DBI(server=server, dbi='sqlite') as db:
+        cat_entries = db.fetchall(
+            'select obsid, slot, starcheck_catalog.id, sc_id, type, mag, mp_starcat_time, dir '
+            'from starcheck_catalog, starcheck_id '
+            'where starcheck_catalog.id is not NULL '
+            'AND starcheck_catalog.sc_id = starcheck_id.id')
+
+    # For this application filter down to guide stars right away
+    ok = np.in1d(cat_entries['type'], ['GUI', 'BOT'])
+    cat_entries = Table(cat_entries[ok])
+    cat_entries.rename_column('id', 'agasc_id')
+    cat_entries = table.join(cat_entries, TIMELINES, keys='dir')
+    # print(len(cat_entries))
+    # cat_entries[-5:].pprint(max_width=-1)
+
+    # Now only accept catalog entries that were actually run by
+    # joining `cat_entries` (which has a lot of candidates that might have been run)
+    # with `starcat_cmds` that were actually run in a timeline.
+
+    ces = table.join(cat_entries, STARCAT_CMDS, keys=['mp_starcat_time', 'timeline_id'])
+    # print(len(ces))
+    # print(ces[-20:])
+    # ces[ces['obsid'] == 19809]
+
+    # Add mag_aca_err column
+
+    filename = os.path.join(os.environ['SKA'], 'data', 'agasc', 'proseco_agasc_1p7.h5')
+    with tables.open_file(filename) as h5:
+        agasc_ids = h5.root.data.col('AGASC_ID')
+        mag_errs = h5.root.data.col('MAG_ACA_ERR') * 0.01
```

Comment:
While fetching stuff from the agasc, maybe it alway makes sense to fetch both mag and mag_aca_err?